tests/test_issue_197_publish_on_change.py

```
@@ -0,0 +1,323 @@
+    ]
+    assert not warning_msgs, f"Unexpected validation warning for valid value: {warning_msgs}"
+
+
+# ---------------------------------------------------------------------------
+# Bug 2 — profile_store snapshot avg_duration fallback
+# ---------------------------------------------------------------------------
+
+
+def test_profile_snapshot_uses_segment_duration_when_avg_duration_zero() -> None:
+    """Bug 2: when profile.avg_duration=0 and cycle.duration=0, the snapshot should
+    estimate duration from the resampled segment (n_samples × used_dt) so that
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)